radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TokenDefinitionsReducer.java

```diff
@@ -58,17 +57,30 @@ public TokenDefinitionsState reduce(TokenDefinitionsState state, TransitionedPar
 				TokenUnitConversions.subunitsToUnits(tokenDefinitionParticle.getGranularity()),
 				tokenSupplyType
 			);
-		} else if (t.getSpinTo() == Spin.UP && (p instanceof MintedTokensParticle || p instanceof BurnedTokensParticle)) {
-			BigInteger mintedOrBurnedAmount = UInt256s.toBigInteger(((Fungible) p).getAmount());
-			BigDecimal change = TokenUnitConversions.subunitsToUnits(
-				(p instanceof BurnedTokensParticle)
-					? mintedOrBurnedAmount.negate()
-					: mintedOrBurnedAmount
-			);
+		} else if (p instanceof MintedTokensParticle || p instanceof BurnedTokensParticle) {
+			if (t.getTransition() == ParticleTransition.N2U) {
+				BigInteger mintedOrBurnedAmount = UInt256s.toBigInteger(((Fungible) p).getAmount());
+				BigDecimal change = TokenUnitConversions.subunitsToUnits(
+					(p instanceof BurnedTokensParticle)
+						? mintedOrBurnedAmount.negate()
+						: mintedOrBurnedAmount
+				);
 
-			TokenDefinitionReference tokenDefinitionReference = p instanceof MintedTokensParticle
-				? ((MintedTokensParticle) p).getTokenDefinitionReference() : ((BurnedTokensParticle) p).getTokenDefinitionReference();
-			return state.mergeSupplyChange(tokenDefinitionReference, change);
+				TokenDefinitionReference tokenDefinitionReference = p instanceof MintedTokensParticle
+					? ((MintedTokensParticle) p).getTokenDefinitionReference() : ((BurnedTokensParticle) p).getTokenDefinitionReference();
+				return state.mergeSupplyChange(tokenDefinitionReference, change);
+			} else if (t.getTransition() == ParticleTransition.U2N) {
```

Comment:
This logic could be simplified a bit by merging the two branches and negating the logic?